CPT: Improve handling of edge case titles in post listing

aduth opened the pull request:
This pull request seeks to...
- Display "(Untitled)" when post lacks title
- Prevent actions from collapsing on itself when title is very long

Before|After
---|---
![Before](https://cloud.githubusercontent.com/assets/1779930/16019550/954702cc-3177-11e6-9db5-1e4524b34163.png)|![After](https://cloud.githubusercontent.com/assets/1779930/16019556/9ccb3e0a-3177-11e6-8bbb-f68f458adc5e.png)

__Testing instructions:__

Verify that if a post has no title, it is shown as "(Untitled)" in the [CPT posts list](http://calypso.localhost:3000/types/post). Similarly, if a post has a very long title that breaks onto a second line, verify that the action buttons do not shrink.

__Follow-up tasks:__

In the context of #5995 where rendered post rows are assigned a fixed height, we should consider whether we'd want the post title to wrap on multiple lines. If not, we'll need to adjust the logic of infinite rendering to track rendered row height.

Test live: https://calypso.live/?branch=fix/cpt-titles

timmyc commented:
I think in stats we use "Untitled" too, but figured I'd check wp-admin and see how testimonials sans-tittle are treated there:

![testimonials_ _ _wordpress](https://cloud.githubusercontent.com/assets/22080/16171296/fe957e74-351f-11e6-96d8-5b57372a3e1c.png)

> In the context of #5995 where rendered post rows are assigned a fixed height, we should consider whether we'd want the post title to wrap on multiple lines.

Seems reasonable to me

Outside of the one question about (Untitled) vs (no title) this LGTM.

timmyc commented:
Happened to find one more random post title edge case which I addressed in 5a31e75 where entities were not being decoded for display, rather than fix in another PR, this seemed quite related to the current topic.

__Before__
![testimonials_ _testingtimmy2_wordpress_com_ _wordpress_com](https://cloud.githubusercontent.com/assets/22080/16171610/f54a463a-3527-11e6-9261-519b84af5c7f.png)

__After__
![testimonials_ _testingtimmy2_wordpress_com_ _wordpress_com](https://cloud.githubusercontent.com/assets/22080/16171616/14535aa8-3528-11e6-81b8-f1d7fe5d1f76.png)

aduth commented:
Thanks @timmyc, and nice catch on the encoded entities in title. This actually raises a good point on whether we should have selectors for "display-ready" posts. Pre-Redux, we'd been relying on `lib/post-normalizer` to correct this. We've run into these issues frequently enough that I think it makes sense. @mtias is exploring a `getNormalizedPost` in #4936 which would decode the post title, taking responsibility away from the component for doing so. If this works out, we might consider exploring something similar for terms.

aduth commented:
Oh, and I changed it to `(no title)`. Not sure this is the best option, but it's [consistent with the pages screen](https://github.com/Automattic/wp-calypso/blob/2849ade13f698dc762ad926676780e30c41405b8/client/my-sites/pages/page.jsx#L258).

mtias commented:
In drafts we use "Untitled".

![image](https://cloud.githubusercontent.com/assets/548849/16224699/74371ab8-37a3-11e6-8557-18e67c5a0d11.png)

aduth commented:
@mtias : Changed text to "Untitled", with same styles pulled from drafts list
@timmyc : Revised your commit to use `getNormalizedPost` introduced in #4936, instead of `decodeEntities` directly

Added long content fade and `nowrap` style to very long titles in 55b8402.

Latest version showing untitled, titles with encoded entities, and very long titles:

![Titles](https://cloud.githubusercontent.com/assets/1779930/16268402/4a18aa94-385c-11e6-956c-a74f33d7adb2.png)
